src/extension.ts

```diff
@@ -21,25 +21,36 @@ export async function activate(ctx: vscode.ExtensionContext): Promise<void> {
 	//
 	// Cursor and VSCode are covered by ms remote, and the only other is windsurf for now
 	// Means that vscodium is not supported by this for now
+	const isTestMode =
+		process.env.NODE_ENV === "test" ||
+		ctx.extensionMode === vscode.ExtensionMode.Test;
+
 	const remoteSSHExtension =
 		vscode.extensions.getExtension("jeanp413.open-remote-ssh") ||
 		vscode.extensions.getExtension("codeium.windsurf-remote-openssh") ||
 		vscode.extensions.getExtension("anysphere.remote-ssh") ||
 		vscode.extensions.getExtension("ms-vscode-remote.remote-ssh");
+
+	let vscodeProposed: typeof vscode = vscode;
+
 	if (!remoteSSHExtension) {
-		vscode.window.showErrorMessage(
-			"Remote SSH extension not found, cannot activate Coder extension",
+		if (!isTestMode) {
+			vscode.window.showErrorMessage(
+				"Remote SSH extension not found, cannot activate Coder extension",
+			);
+			throw new Error("Remote SSH extension not found");
+		}
+		// In test mode, use regular vscode API
```

Comment:
Nice!  I wonder if we can avoid the special testing behavior.  Maybe we should always try to activate, but if we are unable to get this special API we skip the remote registration (and possibly log a warning)?  Similar to what you already have except we always do it, not just in testing.

That would make it nicer for use cases where you have not installed the remote extension yet, for whatever reason, while reducing drift between production and testing.

Reply:
Yeah I suspect this is the kind of behavior we want, so 👍 to logging a warning and handling it gracefully